src/walletConnect/screens/RequestContent.tsx

```diff
@@ -28,6 +30,36 @@ interface Props {
 
 const DAPP_IMAGE_SIZE = 60
 
+export const useDappMetadata = (metadata?: IClientMeta | CoreTypes.Metadata | null) => {
+  const activeDapp = useSelector(activeDappSelector)
+
+  if (!metadata) {
+    // should never happen
+    return {
+      url: '',
+      dappName: '',
+      dappImageUrl: '',
+    }
+  }
+
+  const { url, name, icons } = metadata
+  const dappHostname = new URL(url).hostname
```

Comment:
If the url can't be parsed, this will crash.
Maybe we should cover this with a unit test? :D

Reply:
thank you! i always forget this :( i added unit tests like you suggested